src/commands/profiles/exist.ts

```diff
@@ -0,0 +1,119 @@
+import * as cosmos from "@azure/cosmos";
+import { Command, flags } from "@oclif/command";
+import cli from "cli-ux";
+import * as parse from "csv-parse";
+import * as csvStringify from "csv-stringify";
+import * as fs from "fs";
+import * as transform from "stream-transform";
+import { getCosmosConnection } from "../../utils/azure";
+
+export default class ProfilesExist extends Command {
+  public static description =
+    "Check if, for the given fiscal codes, there are relative profiles or not. It outputs the given csv with a new last column containing true if a profile exists";
+
+  public static flags = {
+    input: flags.string({
+      char: "i",
+      description: "Input file (CSV, with the CF as first column)",
+      required: true
+    }),
+    parallel: flags.integer({
+      char: "p",
+      default: 1,
+      description: "Number of parallel workers to run"
+    })
+  };
+
+  public async run(): Promise<void> {
+    const { flags: parsedFlags } = this.parse(ProfilesExist);
+
+    const inputStream = fs.createReadStream(parsedFlags.input);
+
+    const parser = parse({
+      trim: true,
+      skip_empty_lines: true
+    });
+
+    try {
+      cli.action.start("Retrieving cosmosdb credentials");
+      const { endpoint, key } = await getCosmosConnection(
+        "agid-rg-test",
+        "agid-cosmosdb-test"
+      );
+      cli.action.stop();
+
+      cli.action.start("Querying profiles...");
+      const client = new cosmos.CosmosClient({ endpoint, auth: { key } });
+      const database = client.database("agid-documentdb-test");
+      const container = database.container("profiles");
+
+      // retrieve all fiscalCodes from database
+      const response = container.items.query(
+        "SELECT c.fiscalCode FROM c WHERE c.version = 0",
```

Comment:
better use `COUNT` since [is much cheaper that SELECT](https://microsoft.github.io/AzureTipsAndTricks/blog/tip152.html#get-the-record-count-in-cosmos-db)

Reply:
This select is because I need all fiscalCodes. Then I put them into a Set and check if the given one is that set or not. This code make only 1 query versus cosmosDB.
How can I use count in this scenario?

Reply:
it's one query yes, but the cost model of cosmosdb is based on the amount of data transferred, please check this documents:

* https://docs.microsoft.com/en-us/azure/cosmos-db/request-units

This approach is very inefficient, say we have 10 million profiles in the db, first you query all 10M fiscal codes, then you check whether any of you input fiscal codes (it can be just one) is contained in this 10 million set.

I would suggest to issue a SELECT for each profile - even better, since you know the fiscal code and the version you can hit the specific document and issue the most efficient operation (a READ) - profiles have document IDs equal to `FISCALCODE-VERSION` (see [here](https://github.com/teamdigitale/io-functions-commons/blob/master/src/utils/documentdb_model_versioned.ts#L44))